ls1intum/Artemis
# `Exam mode`: Change test case default visibility to after results release date

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
### Description
* Adjusted the default test case visibility depending on whether the exercise is an exam or course exercise _(and corresponding translations)_
* Fixing issues introduced by #7242 _(console error when loading student exam as instructor, change submitted state displayed twice)_
* Fixed issue in supporting_script introduced in #7379

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Students

1. `instructor`: create a course programming exercise
2. `instructor`: navigate to grading of the exercise and see that the visibility of the test cases is `Always`
3. `instructor`: create an exam programming exercise
4. `instructor`: navigate to grading of the exercise and see that the visibility of the test cases is `After Release Date of Results`
5. `instructor`: make some changes and see that `reset` rests to the expected default state
6. `student`: participate in the course exercise and see that you get immediate feedback
7. `student`: participate in the exam and see that you do only get the info whether your solution compiled

### Testserver States
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [student-exam-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/manage/student-exams/student-exam-detail.component.ts.html) | 90.21% | ✅ ❌ |
| [programming-exercise-grading-tasks-table.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-grading-tasks-table.component.ts.html) | 91.3% | ✅ ❌ |
| [programming-exercise-task.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-task/programming-exercise-task.component.ts.html) | 92.68% | ✅ ❌ |
| [utils.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/util/utils.ts.html) | 98.21% | ✅ ❌ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ProgrammingExercise.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/ProgrammingExercise.html) | 89% | ✅ ❌ |
| [ProgrammingExerciseFeedbackCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseFeedbackCreationService.html) | 94% | ✅ ❌ |
| [ProgrammingExerciseTestCaseService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseTestCaseService.html) | 89% | ✅ ❌ |
| [ProgrammingExerciseTestCaseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/ProgrammingExerciseTestCaseResource.html) | 100% | ✅ |

 

### Screenshots
TODO
